[numpy 2.4.0 deprecation]: TypeError: only 0-dimensional arrays can be converted to Python scalars

Looks like [numpy 2.4.0 has a deprecation](https://github.com/numpy/numpy/releases/tag/v2.4.0):
> Raise TypeError on attempt to convert array with ndim > 0 to scalar
Conversion of an array with ndim > 0 to a scalar was deprecated in NumPy
1.25. Now, attempting to do so raises TypeError. Ensure you extract a
single element from your array before performing this operation.

And [test in nomad-fair repo is failing](https://gitlab.mpcdf.mpg.de/nomad-lab/nomad-FAIR/-/jobs/4485986):
```
                    reference_temperature = None
                    coupling_constant = None
                    if 'nvt' in fix_style or 'npt' in fix_style:
                        thermostat_parameters['thermostat_type'] = 'nose_hoover'
                        if 'temp' in fix:
                            i_temp = np.where(fix == 'temp')[0]
>                           reference_temperature = float(fix[i_temp + 2])  # stop temp
                                                    ^^^^^^^^^^^^^^^^^^^^^^
E                           TypeError: only 0-dimensional arrays can be converted to Python scalars
/usr/local/lib/python3.13/site-packages/atomisticparsers/lammps/parser.py:1194: TypeError
```

--- a/.github/workflows/python-actions.yaml
+++ b/.github/workflows/python-actions.yaml
@@ -1,5 +1,8 @@
 name: install-and-test-workflow
-on: [push]
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
 jobs:
   install-and-test:
     runs-on: ubuntu-latest

--- a/atomisticparsers/gromacs/parser.py
+++ b/atomisticparsers/gromacs/parser.py
@@ -369,7 +369,7 @@ class GromacsMDAnalysisParser(MDAnalysisParser):
                 'MDAnalysis >= 2.0.0 is required for reading force field from tpr. Interactions will not be stored'
             )
             return []
-        gromacs_version = gromacs_version.split('.')[0] if gromacs_version else None
+        gromacs_version = gromacs_version.split('.', maxsplit=1)[0] if gromacs_version else None
         if gromacs_version == '2024':
             self.logger.warning(
                 'Reading force field from tpr not yet supported for Gromacs 2024. Interactions will not be stored'

--- a/atomisticparsers/lammps/parser.py
+++ b/atomisticparsers/lammps/parser.py
@@ -16,38 +16,38 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-import numpy as np
 import os
-from ase import data as asedata
 import re
 
-from nomad.units import ureg
-
+import numpy as np
+from ase import data as asedata
 from nomad.parsing.file_parser import Quantity, TextParser
-from runschema.run import Run, Program
+from nomad.units import ureg
 from runschema.method import (
-    NeighborSearching,
+    AtomParameters,
     ForceCalculations,
     ForceField,
     Method,
     Model,
-    AtomParameters,
+    NeighborSearching,
 )
+from runschema.run import Program, Run
 from runschema.system import AtomsGroup
 from simulationworkflowschema import (
     GeometryOptimization,
     GeometryOptimizationMethod,
     GeometryOptimizationResults,
 )
-from .metainfo.lammps import (
-    x_lammps_section_input_output_files,
-    x_lammps_section_control_parameters,
-)
-from atomisticparsers.utils import MDAnalysisParser, MDParser
 from simulationworkflowschema.molecular_dynamics import (
     get_bond_list_from_model_contributions,
 )
 
+from atomisticparsers.utils import MDAnalysisParser, MDParser
+
+from .metainfo.lammps import (
+    x_lammps_section_control_parameters,
+    x_lammps_section_input_output_files,
+)
 
 re_float = r'[-+]?\d+\.*\d*(?:[Ee][-+]\d+)?'
 re_n = r'[\n\r]'

--- a/atomisticparsers/lammps/parser.py
+++ b/atomisticparsers/lammps/parser.py
@@ -1190,7 +1190,7 @@ class LammpsParser(MDParser):
                     if 'nvt' in fix_style or 'npt' in fix_style:
                         thermostat_parameters['thermostat_type'] = 'nose_hoover'
                         if 'temp' in fix:
-                            i_temp = np.where(fix == 'temp')[0]
+                            i_temp = np.where(fix == 'temp')[0][0]
                             reference_temperature = float(fix[i_temp + 2])  # stop temp
                             coupling_constant = (
                                 float(fix[i_temp + 3]) * integration_timestep
@@ -1242,38 +1242,38 @@ class LammpsParser(MDParser):
                         compressibility = None
                         barostat_type = 'nose_hoover'
                         if 'iso' in fix:
-                            i_baro = np.where(fix == 'iso')[0]
+                            i_baro = np.where(fix == 'iso')[0][0]
                             barostat_parameters['coupling_type'] = 'isotropic'
                             np.fill_diagonal(coupling_constant, float(fix[i_baro + 3]))
                             np.fill_diagonal(reference_pressure, float(fix[i_baro + 2]))
                         else:
                             barostat_parameters['coupling_type'] = 'anisotropic'
                         if 'x' in fix:
-                            i_baro = np.where(fix == 'x')[0]
+                            i_baro = np.where(fix == 'x')[0][0]
                             coupling_constant[0, 0] = float(fix[i_baro + 3])
                             reference_pressure[0, 0] = float(fix[i_baro + 2])
                         if 'y' in fix:
-                            i_baro = np.where(fix == 'y')[0]
+                            i_baro = np.where(fix == 'y')[0][0]
                             coupling_constant[1, 1] = float(fix[i_baro + 3])
                             reference_pressure[1, 1] = float(fix[i_baro + 2])
                         if 'z' in fix:
-                            i_baro = np.where(fix == 'z')[0]
+                            i_baro = np.where(fix == 'z')[0][0]
                             coupling_constant[2, 2] = float(fix[i_baro + 3])
                             reference_pressure[2, 2] = float(fix[i_baro + 2])
                         if 'xy' in fix:
-                            i_baro = np.where(fix == 'xy')[0]
+                            i_baro = np.where(fix == 'xy')[0][0]
                             coupling_constant[0, 1] = float(fix[i_baro + 3])
                             coupling_constant[1, 0] = float(fix[i_baro + 3])
                             reference_pressure[0, 1] = float(fix[i_baro + 2])
                             reference_pressure[1, 0] = float(fix[i_baro + 2])
                         if 'yz' in fix:
-                            i_baro = np.where(fix == 'yz')[0]
+                            i_baro = np.where(fix == 'yz')[0][0]
                             coupling_constant[1, 2] = float(fix[i_baro + 3])
                             coupling_constant[2, 1] = float(fix[i_baro + 3])
                             reference_pressure[1, 2] = float(fix[i_baro + 2])
                             reference_pressure[2, 1] = float(fix[i_baro + 2])
                         if 'xz' in fix:
-                            i_baro = np.where(fix == 'xz')[0]
+                            i_baro = np.where(fix == 'xz')[0][0]
                             coupling_constant[0, 3] = float(fix[i_baro + 3])
                             coupling_constant[3, 0] = float(fix[i_baro + 3])
                             reference_pressure[0, 3] = float(fix[i_baro + 2])
@@ -1289,27 +1289,27 @@ class LammpsParser(MDParser):
                     if fix_style == 'press/berendsen':
                         barostat_type = 'berendsen'
                         if 'iso' in fix:
-                            i_baro = np.where(fix == 'iso')[0]
+                            i_baro = np.where(fix == 'iso')[0][0]
                             barostat_parameters['coupling_type'] = 'isotropic'
                             np.fill_diagonal(coupling_constant, float(fix[i_baro + 3]))
                         elif 'aniso' in fix:
-                            i_baro = np.where(fix == 'aniso')[0]
+                            i_baro = np.where(fix == 'aniso')[0][0]
                             barostat_parameters['coupling_type'] = 'anisotropic'
                             coupling_constant[:3] += 1.0
                             coupling_constant[:3] *= float(fix[i_baro + 3])
                         else:
                             barostat_parameters['coupling_type'] = 'anisotropic'
                         if 'x' in fix:
-                            i_baro = np.where(fix == 'x')[0]
+                            i_baro = np.where(fix == 'x')[0][0]
                             coupling_constant[0] = float(fix[i_baro + 3])
                         if 'y' in fix:
-                            i_baro = np.where(fix == 'y')[0]
+                            i_baro = np.where(fix == 'y')[0][0]
                             coupling_constant[1] = float(fix[i_baro + 3])
                         if 'z' in fix:
-                            i_baro = np.where(fix == 'z')[0]
+                            i_baro = np.where(fix == 'z')[0][0]
                             coupling_constant[2] = float(fix[i_baro + 3])
                         if 'couple' in fix:
-                            i_baro = np.where(fix == 'couple')[0]
+                            i_baro = np.where(fix == 'couple')[0][0]
                             couple = fix[i_baro]
                             if couple == 'xyz':
                                 barostat_parameters['coupling_type'] = 'isotropic'
@@ -1604,7 +1604,7 @@ class LammpsParser(MDParser):
             neighmodify = val[0]  # just use the first instace for now
             neighmodify = np.array([str(i).lower() for i in neighmodify])
             if 'every' in neighmodify:
-                index = np.where(neighmodify == 'every')[0]
+                index = np.where(neighmodify == 'every')[0][0]
                 sec_neighbor_searching.neighbor_update_frequency = int(
                     neighmodify[index + 1]
                 )

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -5,6 +5,7 @@ build-backend = "setuptools.build_meta"
 [project]
 name = "nomad-parser-plugins-atomistic"
 dynamic = ["version"]
+requires-python = ">=3.10"
 description = "Collection of NOMAD parsers for atomistic codes."
 readme = "README.md"
 authors = [{ name = "The NOMAD Authors" }]

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -30,6 +30,11 @@ dependencies = [
 homepage = "https://github.com/nomad-coe/atomistic-parsers"
 
 [project.optional-dependencies]
+infrastructure = [
+    "nomad-lab[infrastructure]>=1.3.10",
+]
+
+[dependency-groups]
 dev = [
     "pytest",
     "pytest-cov",
@@ -37,10 +42,6 @@ dev = [
     'mypy>=1.15',
     'ruff',
     'typing-extensions>=4.12',
-]
-
-infrastructure = [
-    "nomad-lab[infrastructure]>=1.3.10",
 ]
 
 [tool.ruff]

